stdlib/src/builtin/_hash.mojo

```diff
@@ -0,0 +1,179 @@
+# ===----------------------------------------------------------------------=== #
```

Comment:
As we start work to make the generic `HasherTrait` and have multiple hash functions to allow specifying them by the user, it may make sense to make a `hash` top-level module in the `stdlib` and home this new hash type here rather than in `builtin`.  This is still possible to re-export our eventual default hash type/functions now that we have the `prelude` module.

Reply:
I like the idea. Should it be done in this PR or should we make a separate PR for it?

Reply:
Totally fine to do it as a follow up.